core/vm/evm.go

```diff
@@ -454,9 +481,24 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 		err = ErrMaxCodeSizeExceeded
 	}
 
+	// Reject legacy contract deployment from EOF.
+	if err == nil && isInitcodeEOF && !hasEOFMagic(ret) {
+		err = ErrLegacyCode
+	}
```

Comment:
What happens if an eof-contract tries to deploy a legacy -contract? As far as I can tell from this code: 
- the eof-contract `nonce` is increases
- No deployment happens
-  An `0` is returned on the stack, as opposed to an address


Two questions: 
- is that an accurate description? 
- is that correct behaviour? 


cc @chfast

Reply:
And I guess the same questions apply above, re the `initcode` format

Reply:
I think this is OOG-like error instead of returning 0 (same as the `EF` prefix check in the London). So this means the nonce is not increased because state changes are reverted.

This CREATE logic drives me mad.

Reply:
Ok, I was wrong. Your description seems to be correct. Additionally, the caller's nonce is increased.